### PR TITLE
Add vfma/vfnma/vfms/vfnms to arm

### PR DESCRIPTION
**Pull request by aeflores**
This MR addresses https://github.com/NationalSecurityAgency/ghidra/issues/2498

These instructions are part of vfp4, so I added a "define" for it, but I would be happy to do it in some other way.

### CODE REVIEW

**Comment by aeflores** [comment]
The relevant documentation is here: Arm Architecture Reference Manual version G.b https://developer.arm.com/documentation/ddi0487/latest 
* vfma: page 5500
* vfms: page 5514
* vfnma: page 5524
* vfnms:  page 5527

Please let me know if there is something that I can do to help get this reviewed and integrated.
E.g. I could add some tests, but I would need some guidance/references of how processors are tested.

**Comment by aeflores** [comment]
@GhidorahRex any chance of getting this reviewed? These instructions appear a lot in ARM binaries

**Comment by GhidorahRex** [comment]
@aeflores Thanks, this look good. We're planning a few changes to the ARM so I'll try to get this reviewed.

**Comment by GhidorahRex** [comment]
This seems like it's missing a lot of variants. See the latest [DDI0487H](https://developer.arm.com/documentation/ddi0487/latest/) and check the variants for e.g. vfma. These are the A2/T2 VFP4 instructions there should be 3 (f16/f32/f64) of each. With the f16 version, you'll want to use just the lower 2 bytes of each register and zero-extend it into the result.

The A1/T1 versions of these instructions are completely different vectored ops, we will add those separately.

**Comment by aeflores** [comment]
Thanks for the review!

128118b51bd7055af1ecaa3b9963e76736c56a11 adds the 16 bits versions.  I checked that they decode correctly with the following (nonsensical) example:
```
	.text
	.align	1
	.global	main_thumb
	.thumb
	.thumb_func
	.syntax unified
	.fpu vfpv4
	.type	main, %function
main_thumb:
	push	{r4, r7, lr}


	vfma.f16 s13, s14, s15
	vfms.f16 s13, s14, s15
	vfnma.f16 s13, s14, s15
	vfnms.f16 s13, s14, s15
	
	vfma.f32 s13, s14, s15
	vfms.f32 s13, s14, s15
	vfnma.f32 s13, s14, s15
	vfnms.f32 s13, s14, s15
	
	vfma.f64 d5, d6, d7
	vfms.f64 d5, d6, d7
	vfnma.f64 d5, d6, d7
	vfnms.f64 d5, d6, d7
	
	pop	{r4, r7, pc}

	.align	1
	.global	main_arm
	.syntax unified
	.arm
	.type	main, %function	
main_arm:

	push	{r4, r7, lr}

	vfma.f16 s13, s14, s15
	vfms.f16 s13, s14, s15
	vfnma.f16 s13, s14, s15
	vfnms.f16 s13, s14, s15
	
	vfma.f32 s13, s14, s15
	vfms.f32 s13, s14, s15
	vfnma.f32 s13, s14, s15
	vfnms.f32 s13, s14, s15
	
	vfma.f64 d5, d6, d7
	vfms.f64 d5, d6, d7
	vfnma.f64 d5, d6, d7
	vfnms.f64 d5, d6, d7
	
	pop	{r4, r7, pc}
	.size	main_arm, .-main_arm
```
Compiled with `arm-none-eabi-gcc   ex.s -c -o ex`.
Please let me know if the semantics of the 16 bit versions are ok.

File: Ghidra/Processors/ARM/data/languages/ARMneon.sinc [hunk 1]
```diff
@@ -1814,6 +1814,86 @@ extImm: "#"^thv_c0811	is TMode=1 & thv_c0811	{ tmp:1 = thv_c0811; export tmp; }
...
+:vfnma^COND^".f16" Sd,Sn,Sm is ( ( $(AMODE) & COND & c2327=0x1d & c2021=1 & c1011=2 & c0809=1 & c0606=1 & c0404=0 ) |
+						           ($(TMODE_E) & thv_c2327=0x1d & thv_c2021=1 & thv_c1011=2 & thv_c0809=1 & thv_c0606=1 & thv_c0404=0)) & Sm & Sn & Sd
+{
+	Sd = zext(Sd:2 f+ ((0 f- Sn:2) f* Sm:2));
+}
+
+:vfnma^COND^".f32" Sd,Sn,Sm is ( ( $(AMODE) & COND & c2327=0x1d & c2021=1 & c1011=2 & c0809=2 & c0606=1 & c0404=0 ) |
+						           ($(TMODE_E) & thv_c2327=0x1d & thv_c2021=1 & thv_c1011=2 & thv_c0809=2 & thv_c0606=1 & thv_c0404=0)) & Sm & Sn & Sd
+{
+	Sd = Sd f+ ((0 f- Sn) f* Sm);
```
Comment (GhidorahRex) [inline Ghidra/Processors/ARM/data/languages/ARMneon.sinc:1868]: This calculation isn't right. Basedo n the manual, it should be:
`Sd = (f- Sd) f+ ((f- Sn) f* Sm);`
Similarly for the other vfnma instructions.

Comment (aeflores) [reply Ghidra/Processors/ARM/data/languages/ARMneon.sinc]: fixed in 808143cd606e1109bb354ee8873bd9734f5ff4a9

**Pull request merged**